File: core/src/main/java/widget/menu/MenuItemKit.java
package widget.menu;

import android.view.MenuItem;

import org.jetbrains.annotations.NotNull;

/**
 * Created on 2021/4/21
 *
 * @author zsp
 * @desc MenuItem 配套元件
 */
public class MenuItemKit {
    public static MenuItemKit getInstance() {
        return InstanceHolder.INSTANCE;
    }

    /**
     * 显示
     *
     * @param menuItem 菜单条目
     */
    public void show(@NotNull MenuItem menuItem) {
        menuItem.setVisible(true);
        menuItem.setEnabled(true);
    }

    /**
     * 隐藏
     *
     * @param menuItem 菜单条目
     */
    public void hide(@NotNull MenuItem menuItem) {
        menuItem.setVisible(false);
        menuItem.setEnabled(false);
    }

    private static final class InstanceHolder {
        static final MenuItemKit INSTANCE = new MenuItemKit();
    }
}

File: core/src/main/java/widget/menu/MenuItemKit.kt
package widget.menu

import android.view.MenuItem

/**
 * Created on 2026/4/8.
 * @author 郑少鹏
 * @desc MenuItem 配套元件
 */
class MenuItemKit {
    companion object {
        /**
         * 显示
         *
         * @param menuItem 菜单条目
         */
        fun show(menuItem: MenuItem) {
            menuItem.apply {
                isVisible = true
                isEnabled = true
            }
        }

        /**
         * 隐藏
         *
         * @param menuItem 菜单条目
         */
        fun hide(menuItem: MenuItem) {
            menuItem.apply {
                isVisible = false
                isEnabled = false
            }
        }
    }
}